.github/workflows/pr-test-suite.yml

```diff
@@ -116,7 +116,7 @@ jobs:
         cache: false
     - uses: Swatinem/rust-cache@v2
       with:
-        key: ${{ runner.os }}-build
+        prefix-key: ${{ runner.os }}-build
```

Comment:
Note that I updated some of these to use prefix-key instead of key, because based on the docs for the action, it automatically prepends the cache key with the name of the job. So it never actually shares the cache between the 3 jobs that have the key `${{ runner.os }}-build`. Not strictly required but should be useful to avoid excess cache writes?